modules/rostests/kmtests/ntos_io/IoDeviceInterface.c

```diff
@@ -1,292 +1,221 @@
-
-    RtlInitEmptyUnicodeString(&SymbolicLinkName, NULL, 0);
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_INVALID_PARAMETER);
-
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_INVALID_PARAMETER);
-
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??\\");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_INVALID_PARAMETER);
-
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??\\\\");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_INVALID_PARAMETER);
-
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??\\{aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa}");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_INVALID_PARAMETER);
-
-    /* Valid prefix & GUID, invalid device node */
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??\\X{aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa}");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_OBJECT_NAME_NOT_FOUND);
-
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\\\?\\X{aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa}");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_OBJECT_NAME_NOT_FOUND);
-
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??\\X{aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa}\\");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_OBJECT_NAME_NOT_FOUND);
-
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??\\#{aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa}");
-    KmtStartSeh()
-        Status = IoSetDeviceInterfaceState(&SymbolicLinkName, TRUE);
-    KmtEndSeh(STATUS_SUCCESS)
-    ok_eq_hex(Status, STATUS_OBJECT_NAME_NOT_FOUND);
-
-    /* Must not read past the buffer */
-    RtlInitUnicodeString(&SymbolicLinkName, L"\\??\\#{aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa}");
-    BufferSize = SymbolicLinkName.Length;
-    Buffer = KmtAllocateGuarded(BufferSize);
-    if (!skip(Buffer != NULL, "Failed to allocate %lu bytes\n", BufferSize))
+    PWSTR SymbolicLinkList, SymbolicLinkListPtr;
+    UNICODE_STRING GuidString;
+
+    RtlStringFromGUID(guid, &GuidString);
+
+    Status = IoGetDeviceInterfaces(guid, NULL, DEVICE_INTERFACE_INCLUDE_NONACTIVE, &SymbolicLinkList);
+
+    ok(NT_SUCCESS(Status), "IoGetDeviceInterfaces failed with status 0x%X for '%wZ'\n", (unsigned int)Status, &GuidString);
```

Comment:
What's the point of the type cast here?

Reply:
Oh, looks like I didn't check this part of the code yet, since it was wrote by @learn-more. But I'm sure that this cast is just not needed and can be safely removed.

Reply:
probably gcc crying about crap

Reply:
Thanks. Just removed it, and at least GCC build still doesn't fail for me. So it indeed seems to be not needed.

Reply:
Fixed. :ok_hand: